cmd/kubelet/app/server.go

```diff
@@ -305,7 +305,23 @@ func initConfigz(kc *componentconfig.KubeletConfiguration) (*configz.Config, err
 	return cz, err
 }
 
+// This serializes the values in the s.Experimental struct (set via flags)
+// to the JSON string s.KubeletConfiguration.Experimental
+func mirrorExperimentalToKubeletConfiguation(s *options.KubeletServer) error {
+	str, err := componentconfig.ExperimentalKubeletConfigurationToString(&s.Experimental)
+	if err != nil {
+		return err
+	}
+	s.KubeletConfiguration.Experimental = str
+	return nil
+}
+
 func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
+	// Immediately serialize s.Experimental into s.KubeletConfiguration.Experimental
+	// so that values from flags are represented in the Experimental string.
+	// This ensures experimental configuration will be visible on /configz, for example.
+	mirrorExperimentalToKubeletConfiguation(s)
```

Comment:
A general nit. This function has grown quite a bit. Why not extract all the config related logic to separate functions and possibly place them in a separate file?